### PR TITLE
Fix `IForestOutlierTransform` `features_to_ignore` parameter fail on `target`

### PR DESCRIPTION
**Pull request by brsnw250**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: etna/analysis/outliers/isolation_forest_outliers.py [hunk 1]
```diff
@@ -35,7 +35,9 @@ def _select_features(
             "Changing the defaults there should be exactly one option set: features_to_use or features_to_ignore"
         )
     features_to_ignore = list(set(features_to_ignore) - {in_column})
-    df = df.drop(columns=features_to_ignore, level="feature")
+    if len(features_to_ignore) > 0:
```
Comment (d-a-bunin) [inline etna/analysis/outliers/isolation_forest_outliers.py:38]: Let's add which versions of pandas fail here. We added such comments in other similar places before.

File: tests/test_transforms/test_decomposition/test_model_based.py [hunk 2]
```diff
@@ -337,3 +337,32 @@ def test_stride_transform(forward_stride_datasets, decompose_model, context_size
 
     assert not transformed.df.iloc[context_size:10].isna().any().any()
     assert transformed.df.iloc[10:].isna().all().any()
+
+
+@pytest.mark.parametrize("features_to_ignore", (["target"], ["regressor_1"], ["target", "regressor_1"]))
+def test_repetitive_call_before_pipeline_with_ignore(outliers_solid_tsds, features_to_ignore):
```
Comment (d-a-bunin) [inline tests/test_transforms/test_decomposition/test_model_based.py:343]: Can you explain why do we need such test?

**Review by d-a-bunin — CHANGES_REQUESTED** [review]

File: tests/test_transforms/test_decomposition/test_model_based.py (hunk 2 above)
Comment (brsnw250) [reply tests/test_transforms/test_decomposition/test_model_based.py:343]: To cover [this](https://github.com/etna-team/etna/actions/runs/10633068062/job/29477322573#step:6:5595) case

Comment (d-a-bunin) [reply tests/test_transforms/test_decomposition/test_model_based.py:343]: Can you explain how it arises here?

File: tests/test_transforms/test_decomposition/test_model_based.py [hunk 3]
```diff
@@ -337,3 +337,38 @@ def test_stride_transform(forward_stride_datasets, decompose_model, context_size
...
+    """Problem arises when decomposition transform is applied the second time to the same `TSDataset`.
```
Comment (d-a-bunin) [inline tests/test_transforms/test_decomposition/test_model_based.py:344]: An empty line should be added here.

I'll approve, and you could do this if you think it is necessary.

**Review by d-a-bunin — APPROVED** [review]

**Pull request merged**